tfhe/benches/shortint/bench.rs

```diff
@@ -304,6 +306,39 @@ fn programmable_bootstrapping(c: &mut Criterion) {
     bench_group.finish();
 }
 
+fn server_key_from_compressed_key(c: &mut Criterion) {
+    let mut bench_group = c.benchmark_group("uncompress_key");
+    bench_group
+        .sample_size(10)
+        .measurement_time(std::time::Duration::from_secs(60));
+
+    for param in SERVER_KEY_BENCH_PARAMS {
+        let param: PBSParameters = param.into();
+        let keys = KEY_CACHE.get_from_param(param);
+        let sks_compressed = CompressedServerKey::new(&keys.client_key());
+
+        let bench_id = format!("shortint::uncompress_key::{}", param.name());
+
+        bench_group.bench_function(&bench_id, |b| {
+            b.iter(|| {
+                let _ = ServerKey::from(sks_compressed.clone());
```

Comment:
the clone is going to give wrong results as the clone operation is going to appear in the global time of the benchmark
we need an iter with input preparation I believe, integer benchmarks have it IIRC or maybe it's also in that file

Reply:
perfect